str0m/src/media/register.rs

```diff
@@ -261,47 +328,57 @@ impl ReceiverRegister {
         let stop = *self.max_seq - MISORDER_DELAY;
 
         if stop < start {
-            return None;
+            return vec![];
         }
 
+        let mut nacks = vec![];
         let mut first_missing = None;
         let mut bitmask = 0;
 
-        // this might be changed again if we end the loop early
-        self.nack_check_from = stop.into();
-
         for i in start..stop {
-            let j = (i % self.bits.len() as u64) as usize;
+            let j = self.packet_index(i);
 
-            // zero the bit and know if we received the packet.
-            let did_receive = self.bits.replace(j, false);
+            let should_nack = self.packet_status[j].should_nack();
 
             if let Some(first) = first_missing {
-                if !did_receive {
+                if should_nack {
                     let o = (i - (first + 1)) as u16;
                     bitmask |= 1 << o;
+                    self.packet_status[j].nack_count += 1;
                 }
 
                 if i - first == 16 {
-                    // early break because we can report max 17 valus each report.
-                    // reset check_from.
-                    self.nack_check_from = (i + 1).into();
-                    break;
+                    nacks.push(NackEntry {
+                        pid: (first % u16::MAX as u64) as u16,
+                        blp: bitmask,
+                    });
+                    bitmask = 0;
+                    first_missing = None;
                 }
-            } else if !did_receive {
+            } else if should_nack {
+                self.packet_status[j].nack_count += 1;
                 first_missing = Some(i);
             }
         }
 
-        first_missing.map(|first| Nack {
-            sender_ssrc: 0.into(),
-            ssrc: 0.into(), // changed when sending
-            reports: NackEntry {
+        if let Some(first) = first_missing {
+            nacks.push(NackEntry {
                 pid: (first % u16::MAX as u64) as u16,
                 blp: bitmask,
-            }
-            .into(),
-        })
+            });
+        }
+
+        let reports = ReportList::lists_from_iter(nacks).into_iter();
```

Comment:
I would like `lists_from_iter` to return another iterator, this is possible with `itertools`'s `chunks` but I didn't feel worth it to add a dependency just for that

Reply:
slice has chunks(), would that help ?

Reply:
Hmmm I wanted to take `IntoIterator` as the argument but since I happen to have a vec here maybe I should just use that